File: packages/core/src/context/ContextEngine.ts
import { ContextSignal, ContextSignalType } from "./ContextSignal";
import { ContextSnapshot } from "./ContextSnapshot";

export class ContextEngine {
  private signals: Map<string, ContextSignal>;

  constructor() {
    this.signals = new Map();
  }

  addSignal(signal: ContextSignal): void {
    const key = `${signal.type}:${signal.key}`;
    this.signals.set(key, signal);
  }

  getSignal(type: ContextSignalType, key: string): ContextSignal | undefined {
    return this.signals.get(`${type}:${key}`);
  }

  snapshot(userId: string): ContextSnapshot {
    return {
      userId,
      timestamp: new Date().toISOString(),
      signals: Array.from(this.signals.values()),
    };
  }

  clear(): void {
    this.signals.clear();
  }
}

export function createContextEngine(): ContextEngine {
  return new ContextEngine();
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/context/ContextEngine.kt
package ai.mobileagentos.compose.context

class ContextEngine {
    private val signals: MutableMap<String, ContextSignal> = mutableMapOf()
    
    fun addSignal(signal: ContextSignal) {
        val key = "${signal.type}:${signal.key}"
        signals[key] = signal
    }
    
    fun getSignal(type: ContextSignalType, key: String): ContextSignal? {
        return signals["${type}:${key}"]
    }
    
    fun snapshot(userId: String): ContextSnapshot {
        return ContextSnapshot(
            userId = userId,
            timestamp = System.currentTimeMillis().toString(),
            signals = signals.values.toList()
        )
    }
    
    fun clear() {
        signals.clear()
    }
}